src/bank.rs

```diff
@@ -310,21 +375,56 @@ impl Bank {
                 .collect();
             // There is no way to predict what contract will execute without an error
             // If a fee can pay for execution then the contract will be scheduled
-            self.reserve_signature_with_last_id(&tx.signature, &tx.last_id)?;
+            let err =
+                Self::reserve_signature_with_last_id(last_ids_sigs, &tx.last_id, &tx.signature);
+            err?;
             called_accounts[0].tokens -= tx.fee;
             Ok(called_accounts)
         }
     }
+    fn lock_accounts(&self, txs: &[Transaction]) -> Vec<Result<()>> {
```

Comment:
This should have a `#[must_use]` annotation to tell rust to tell users to dig down to those Results.